File: Sistema-Ganimedes.Domain/Entities/Usuario.cs
using Sistema_Ganimedes.Domain.Enums;
using System.Text.Json.Serialization;

namespace Sistema_Ganimedes.Domain.Entities
{
    public class Usuario
    {
        [JsonPropertyName("nusp")]
        public string nusp { get; set; }

        [JsonPropertyName("nome")]
        public string nome { get; set; }

        [JsonPropertyName("email")]
        public string email { get; set; }

        [JsonPropertyName("link_lattes")]
        public string linkLattes { get; set; }

        [JsonPropertyName("dt_atualizacao_lattes")]
        public DateOnly dtAtualizacaoLattes { get; set; }

        [JsonPropertyName("perfil")]
        public string perfil {  get; set; }
        public TipoUsuario tipo_usuario { get; set; }
    }
}

File: ganimedes_authentication_service/src/main/java/com/ganimedes/ganimedes_authentication_service/model/Usuario.java
package com.ganimedes.ganimedes_authentication_service.model;

import com.ganimedes.ganimedes_authentication_service.model.enums.TipoUsuario;
import jakarta.persistence.*;
import java.time.LocalDate;

@Entity
@Table(name = "usuario")
public class Usuario {

    @Id
    @Column(name = "nusp", nullable = false, unique = true)
    private String nUsp;

    @Column(name = "nome", nullable = false)
    private String nome;

    @Column(name = "email", nullable = false, unique = true)
    private String email;

    @Column(name = "password", nullable = false)
    private String password;

    @Column(name = "link_lattes")
    private String linkLattes;

    @Column(name = "dt_atualizacao_lattes")
    private LocalDate dtAtualizacaoLattes;

    @Column(name = "perfil", nullable = false)
    private String tipoUsuario;

    public String getNUsp() {
        return nUsp;
    }

    public void setNUsp(String nUsp) {
        this.nUsp = nUsp;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getLinkLattes() {
        return linkLattes;
    }

    public void setLinkLattes(String linkLattes) {
        this.linkLattes = linkLattes;
    }

    public LocalDate getDtAtualizacaoLattes() {
        return dtAtualizacaoLattes;
    }

    public void setDtAtualizacaoLattes(LocalDate dtAtualizacaoLattes) {
        this.dtAtualizacaoLattes = dtAtualizacaoLattes;
    }

    public String getTipoUsuario() {
        return tipoUsuario;
    }

    public void setTipoUsuario(String tipoUsuario) {
        this.tipoUsuario = tipoUsuario;
    }
}

